watertap/unit_models/ion_exchange_0D.py

```diff
@@ -0,0 +1,1352 @@
+        )
+
+        self.regen_pump_power = Var(
+            initialize=4e-6, units=pyunits.kW, doc="Regen pump power [kW]"
+        )
+
+        self.bw_pump_power = Var(
+            initialize=4e-6, units=pyunits.kW, doc="Backwash pump power [kW]"
+        )
+
+        self.rinse_pump_power = Var(
+            initialize=4e-6, units=pyunits.kW, doc="Rinse pump power [kW]"
+        )
+
+        self.pump_efficiency = Var(
+            initialize=0.8,
+            units=pyunits.dimensionless,
+            bounds=(0, 1),
+            doc="Pump efficiency",
+        )
+
+        # Fix variables that should be fixed
+        self.resin_diam.fix()
+        self.resin_bulk_dens.fix()
+        self.bed_porosity.fix()
+        self.dimensionless_time.fix()
+        self.lh.fix()
+        self.regen_dose.fix()
+        self.regen_ww.fix()
+        self.regen_sg.fix()
+        self.t_bw.fix()
+        self.bw_rate.fix()
+        self.rinse_bv.fix()
+        self.pump_efficiency.fix()
+
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = True  # inlet block is an inlet
+
+        self.properties_in = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of inlet",
+            default=tmp_dict,
+        )
+
+        # Add outlet and waste block
+        tmp_dict["defined_state"] = False  # outlet and waste block is not an inlet
+        self.properties_out = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of outlet",
+            default=tmp_dict,
+        )
+
+        self.properties_waste = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of waste",
+            default=tmp_dict,
+        )
+
+        # Add ports - oftentimes users interact with these rather than the state blocks
+        self.add_port(name="inlet", block=self.properties_in)
+        self.add_port(name="outlet", block=self.properties_out)
+        self.add_port(name="waste", block=self.properties_waste)
+
+        # Add constraints
+        # =========== EQUILIBRIUM ===========
+
+        @self.Constraint(
+            ion_set,
+            doc="Separation factor calc",
+        )
+        def eq_sep_factor(b, j):
+            return b.R_eq[j] == 1 / b.K_eq[j]
+
+        @self.Constraint(
+            ion_set,
+            doc="Langmuir isotherm",
+        )
+        def eq_langmuir(b, j):
+            if j == target_ion:
+                return b.R_eq[j] == (
+                    b.c_norm[j] * (1 - b.resin_eq_capacity / b.resin_max_capacity)
+                ) / (b.resin_eq_capacity / b.resin_max_capacity * (1 - b.c_norm[j]))
+            else:
+                return Constraint.Skip
```

Comment:
Will this unit model be limited to only removing a single target ion? Can this be further generalized for a set of target ions?

Reply:
So a fixed-bed process can be modeled as a batch process whose control volume moves at the speed of the fluid, so I would assume that almost anything that would apply to the batch would also apply to the fixed bed.

Reply:
Right, but I haven't seen anything in literature suggesting you can apply the batch equilibrium relationships to fixed-bed transitions; I could check again. Either way, I would prefer to get something merged for a single component before trying to expand it to multiple components.

Reply:
If there is a plan to expand to multi-component at some point, I would recommend going ahead and having the logic built into the unit model for dealing with a list of target ions, then adding an error check in the build or at specific constraints to report an error if the list is >1. It will be a lot more work to add in multiple species later if you don't plan for it now. 

Specifically, I would make the target ions config option to accept a list of strings (where each string is presumably the name of a target ion). Then, towards the front of the build check to see if the list is larger than 1 and report an error. It may also be worth checking to make sure that the name in the list actually exists as a component/solution/ion name from the property package. 

In the constraint (like this one), instead of checking if `j == target`, you could do `if j in target_list`. Then, that function will get applied whenever one of the j components is found in the list of targets. 

That is a more generalized way to build it that will also allow for easily extending later.